gadget/device/encrypt_test.go

```diff
@@ -105,3 +105,49 @@ func (s *deviceSuite) TestLocations(c *C) {
 	c.Check(device.TpmLockoutAuthUnder(dirs.SnapFDEDirUnderSave(dirs.SnapSaveDir)), Equals,
 		"/var/lib/snapd/save/device/fde/tpm-lockout-auth")
 }
+
+func (s *deviceSuite) TestStampSealedKeysRunthrough(c *C) {
+	root := c.MkDir()
+
+	for _, tc := range []struct {
+		mth      device.SealingMethod
+		expected string
+	}{
+		{device.SealingMethodLegacyTPM, ""},
+		{device.SealingMethodTPM, "tpm"},
+		{device.SealingMethodFDESetupHook, "fde-setup-hook"},
+	} {
+		err := device.StampSealedKeys(root, tc.mth)
+		c.Assert(err, IsNil)
+
+		mth, err := device.SealedKeysMethod(root)
+		c.Assert(err, IsNil)
+		c.Check(tc.mth, Equals, mth)
+
+		content, err := ioutil.ReadFile(filepath.Join(root, "/var/lib/snapd/device/fde/sealed-keys"))
+		c.Assert(err, IsNil)
+		c.Check(string(content), Equals, tc.expected)
+	}
+}
+
+func (s *deviceSuite) TestStampSealedKeysMissing(c *C) {
```

Comment:
nitpick^1000: I'd name the test after the main function being tested, so `TestSealedKeysMethodWithMissingStamp`.